lib/originals-helper.js

```diff
@@ -56,6 +56,75 @@ function getModuleNameForType(type) {
   }
 }
 
+/**
+ * @param {Node} node - a ts-morph AST node
+ * @param {string} property - the property to find on some module
+ *
+ * @return {string} the get-originals module name corresponding to the node's
+ * type. (This is sometimes different from the TypeScript type name.)
+ */
+function getModuleNameForNodeFromPrototypeChain(node, property) {
+  return getModuleNameForTypeFromPrototypeChain(node.getType(), property);
+}
+
+/**
+ * @param {Type} type - a ts-morph type
+ * @param {string} property - the property to find on some module
+ *
+ * @return {string} whether the property exists on the given type
+ */
+function checkPropertyOnType(type, property) {
+  for (const symbol of type.getProperties()) {
+    if (property === symbol.getName()) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * @param {Type} type - a ts-morph type
+ * @param {string} property - the property to find on some module
+ *
+ * @return {string} the get-originals module name corresponding to the
+ * type. (This is sometimes different from the TypeScript type name.)
+ */
+function getModuleNameForTypeFromPrototypeChain(type, property) {
+  // Need to use BaseTypeOfLiteralType to catch the type of primitives
+  switch (type.getBaseTypeOfLiteralType().getText()) {
+    case 'string': {
+      return 'String';
+    }
+    case 'number': {
+      return 'Number';
+    }
+    case 'boolean': {
+      return 'Boolean';
+    }
+  }
+
+  // Find the deepest reference to the property in the prototype chain
+  let inspectType = type;
+  let latestFound;
+  while (inspectType !== undefined) {
+    if (checkPropertyOnType(inspectType, property)) {
+      latestFound = inspectType;
+    } else if (latestFound !== inspectType) {
```

Comment:
Would it be possible to remove this else if? It seems like line 120 should cover it?

Reply:
This is an optimization that will break you out of the prototype chain once you've found the most "super" location. I could leave the return to line 120 but I figured if I could add this break I should.

I'm also noticing now that this will fail if the property does not exist on `type`, since latest found will be `undefined`, but that's a case that will only happen if you write code that accesses a property on a global that doesn't exist, which is an error anyway.